### PR TITLE
C# - Swagger Updates

### LINKED ISSUE #48: Update swagger spec for Healthz
## Description:
- /healthz swagger docs still references "Healthy", "Degraded", "Unhealthy"
- /healthz/ietf and /healthz/dotnet have strange values for example value and schema
- Probably worth meeting to discuss expectations all up for swagger docs, maybe combine with https://github.com/retaildevcrews/helium/issues/243?

### LINKED ISSUE #54: Actor: Validate Helium API Parameters as per specifications
## Description
Currently Helium C# API Parameters are not following standard specifications.

## Expected Behavior
- The goal is to apply commons parameter validations for actors
https://github.com/retaildevcrews/helium/blob/api_specs/docs/HeliumParamSpecs.md
- On API calls, the responses for valid and invalid scenarios should reflect rules specified in the specifications.

### LINKED ISSUE #55: Movies: Validate Helium API Parameters as per specifications
## Description
Currently Helium C# API Parameters for Movies are not following standard specifications.

## Expected Behavior
- The goal is to apply commons parameter validations for movies
https://github.com/retaildevcrews/helium/blob/master/docs/ParameterValidation.md

- On API calls, the responses for valid and invalid scenarios should reflect rules specified in the specifications.

### PR DESCRIPTION
**Pull request by bartr**
## Type of PR

- [X] Documentation changes
- [X] Code changes
- [x] Test changes
- [ ] CI-CD changes
- [ ] GitHub Template changes

## Purpose of PR
- update packages
- consolidate fxcop overrides into one file
- disable swagger auto generation of json
- add swagger json file
- add parameter validation

## Issues Closed or Referenced

- Closes #48 
- Closes #54
- Closes #55

### CODE REVIEW

File: src/app/Controllers/ParameterValidator.cs [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+            if (!Common(query, q, pageNumber, pageSize, out message))
+            {
+                return false;
+            }
+
+            // validate genre
+            if (query.ContainsKey("genre"))
+            {
+                if (genre == null || genre.Length < 3 || genre.Length > 50)
+                {
```
Comment (aflinchb) [inline src/app/Controllers/ParameterValidator.cs:96]: In parameter validation doc, genre length max is 20. #Closed

File: src/app/Controllers/MoviesController.cs [hunk 2]
```diff
@@ -43,32 +44,30 @@ public MoviesController(ILogger<MoviesController> logger, IDAL dal)
         /// <param name="pageSize">page size (1000 max)</param>
         /// <response code="200">JSON array of Movie objects or empty array if not found</response>
         [HttpGet]
-        [Produces("application/json")]
         [ProducesResponseType(typeof(Movie[]), 200)]
-        public async Task<IActionResult> GetMoviesAsync([FromQuery]string q = "", [FromQuery] string genre = "", [FromQuery] int year = 0, [FromQuery] double rating = 0, [FromQuery] bool topRated = false, [FromQuery] string actorId = "", [FromQuery] int pageNumber = 1, [FromQuery] int pageSize = Constants.DefaultPageSize)
+        [ProducesResponseType(typeof(string), 400)]
+        public async Task<IActionResult> GetMoviesAsync([FromQuery]string q = null, [FromQuery] string genre = null, [FromQuery] int year = 0, [FromQuery] double rating = 0, [FromQuery] bool topRated = false, [FromQuery] string actorId = null, [FromQuery] int pageNumber = 1, [FromQuery] int pageSize = Constants.DefaultPageSize)
```
Comment (aflinchb) [inline src/app/Controllers/MoviesController.cs]: As part of parameter validation work, I believe we agreed to remove "topRated"

Comment (aflinchb) [inline src/app/Controllers/MoviesController.cs]: I noticed that this method now has [ProducesResponseType(typeof(Movie[]...] *and*  [ProducesResponseType(typeof(string[]...], but no changes to the attributes in the ActorsController.  I think these are only used by the auto-generated Swagger anyway (could be wrong).  If we are going to provide the option to not use the json file and use the auto-generated one, we should keep the comments consistent.

Comment (bartr) [reply src/app/Controllers/MoviesController.cs]: updated

Comment (bartr) [reply src/app/Controllers/MoviesController.cs]: Removed and added note to board to remove from other repos

**Review by aflinchb — APPROVED** [review]

**Review by evillgenius75 — APPROVED** [review]
LGTM

File: src/app/Controllers/MoviesController.cs (hunk 2 above)
Comment (aflinchb) [reply src/app/Controllers/MoviesController.cs]: testing to see how this looks in github (this is from Code Flow)

---
In reply to: [386680191](https://github.com/retaildevcrews/helium-csharp/pull/60#discussion_r386680191) [](ancestors = 386680191)

**Pull request merged**